cli/azd/pkg/pipeline/pipeline_manager.go

```diff
@@ -906,14 +945,31 @@ func (pm *PipelineManager) promptForCiFiles(ctx context.Context, pipelineProvide
 		}
 
 		if !osutil.FileExists(providerPaths.yml) {
-			embedFilePath := fmt.Sprintf("pipeline/.%s/azure-dev.yml", pipelineProvider)
-			if infraProvider == "terraform" {
-				embedFilePath = fmt.Sprintf("pipeline/.%s/azure-dev-tf.yml", pipelineProvider)
+			embedFilePath := fmt.Sprintf("pipeline/.%s/azure-dev.ymlt", props.CiProvider)
+			tmpl, err := template.
+				New("azure-dev.yml").
+				Option("missingkey=error").
+				ParseFS(resources.PipelineFiles, embedFilePath)
+			if err != nil {
+				return fmt.Errorf("parsing embedded file %s: %w", embedFilePath, err)
 			}
-			contents, err := resources.PipelineFiles.ReadFile(embedFilePath)
+			builder := strings.Builder{}
+			err = tmpl.Execute(&builder, struct {
+				BranchName          string
+				FedCredLogIn        bool
+				InstallDotNetAspire bool
+				ClientSecretLogIn   bool
+			}{
+				BranchName:          props.BranchName,
```

Comment:
NIT: I don't think that `FedCredLogIn` and `ClientSecretLogIn` should ever both be true or both be false, right?  You must pick one or the other.  It might be nice to encode that in the context object (by only having a single property) and then have the template just do an if/else?